src/composables/maxAmount.ts

```diff
@@ -0,0 +1,129 @@
+  validateTipUrl,
+  checkAensName,
+  handleUnknownError,
+  watchUntilTruthy,
+} from '../popup/utils';
+import { IAccount } from '../types';
+
+export interface UseMaxAmountOptions {
+  store: Store<any>
+  formModel: any,
+}
+
+export function useMaxAmount({ store, formModel }: UseMaxAmountOptions) {
+  const fee = ref(new BigNumber(0));
+  const selectedTokenBalance = ref(new BigNumber(0));
+  const tokenInstance = ref<any>(null);
+  const nonce = ref(0);
+  const selectedAssetDecimals = ref(0);
+
+  const balance = rxJsObservableToVueState(store.state.observables.balance);
+  const balanceCurrency = rxJsObservableToVueState(store.state.observables.balanceCurrency);
+  const account = computed<IAccount>(() => store.getters.account);
+  const max = computed(() => {
+    if (balance.value && formModel.value?.selectedAsset?.contractId === AETERNITY_CONTRACT_ID) {
+      const _max = balance.value.minus(fee.value);
+      return (_max.isPositive() ? _max : 0).toString();
+    }
+    return selectedTokenBalance.value.toString();
+  });
+
+  watch(
+    () => formModel.value,
+    async (val) => {
+      if (!val?.selectedAsset) return;
+      await watchUntilTruthy(() => store.state.sdk);
+      const { sdk } = store.state;
+
+      if (val.selectedAsset.contractId !== AETERNITY_CONTRACT_ID) {
+        if (!tokenInstance.value
+          || tokenInstance.value.deployInfo.address !== val.selectedAsset.contractId) {
+          tokenInstance.value = await store.state.sdk.getContractInstance({
+            source: FUNGIBLE_TOKEN_CONTRACT,
+            contractAddress: val.selectedAsset.contractId,
+          });
+        }
+        selectedAssetDecimals.value = val.selectedAsset.decimals;
+      }
+
+      if (val.selectedAsset.contractId !== AETERNITY_CONTRACT_ID
+        || (val.address && !checkAensName(val.address) && validateTipUrl(val.address))) {
+        fee.value = calculateFee(
+          SCHEMA.TX_TYPE.contractCall, {
+            ...sdk.Ae.defaults,
+            ttl: 0,
+            nonce: nonce.value + 1,
+            amount: (new BigNumber(val.amount > 0 ? val.amount : 0)).shiftedBy(MAGNITUDE),
+            callerId: account.value.address,
+            contractId: validateTipUrl(val.address)
+              ? STUB_CONTRACT_ADDRESS : val.selectedAsset.contractId,
+          },
+        );
+        return;
+      }
+      const minFee = new BigNumber(TxBuilder.calculateMinFee('spendTx', {
+        gas: sdk.Ae.defaults.gas,
+        params: {
+          ...sdk.Ae.defaults,
+          senderId: account.value.address,
+          recipientId: account.value.address,
+          amount: new BigNumber(val.amount > 0 ? val.amount : 0).shiftedBy(MAGNITUDE),
+          ttl: 0,
+          nonce: nonce.value + 1,
+          payload: '',
+        },
+      })).shiftedBy(-MAGNITUDE);
+      if (!minFee.isEqualTo(fee.value)) fee.value = minFee;
+    },
+    { deep: true },
+  );
+
+  onMounted(() => {
+    executeAndSetInterval(async () => {
```

Comment:
This function returns the `setInterval` handle. So the this mixin should also have the `onBeforeUnmount` where the timeouts should be removed.